src/lib.rs

```diff
@@ -251,6 +240,28 @@ cfg_if! {
     }
 }
 
+/// Register a function to be invoked by `getrandom` on custom targets. This
+/// function will only be invoked on targets not supported by `getrandom`. This
+/// prevents crate dependencies from either inadvertently or maliciously
+/// overriding the secure RNG implementations in `getrandom`.
+///
+/// *This API requires the following crate features to be activated: `custom`*
+#[macro_export]
+#[cfg(feature = "custom")]
+macro_rules! register_custom_getrandom {
```

Comment:
Note that Edition 2018 macro rules enable us to place this macro in the `custom` sub-module, then `pub use` it. We require Rustc >= 32 so I don't see any problems doing this, though the keeping it here is also fine (just slightly messier).

Reply:
So we have to be careful, as the marco needs to always be exported, even if the custom implementation isn't used. Otherwise adding support for a target becomes a breaking change.

To get this working I did:

In `src/lib.rs`:
```rust
#[cfg(feature = "custom")]
mod custom;

// ...

cfg_if! {
    // ...
    } else if #[cfg(feature = "custom")] {
        use custom as imp;
    } else {
    // ...
}
```

In `src/custom.rs`:
```rust
#[macro_export]
macro_rules! register_custom_getrandom {
    // ...
}

#[allow(dead_code)]
pub fn getrandom_inner(dest: &mut [u8]) -> Result<(), Error> {
    // ...
}
```

Does this seem reasonable?

Reply:
Seems I mis-remembered how macros get exported ([silly example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7ad67c2ad78ff237d79545259c6a3bbc)). So this is as good as we can get.

Any idea why we need `#[alllow(dead_code)]` now?

Reply:
`getrandom_inner` isn't always called. If you have `custom` selected on a supported target, you'll get a dead code warning.

Reply:
Sorry, I get it now.

So, this change results in slightly more code overall, but does but both parts of this `extern "C"` interface within the same module and make `lib.rs` smaller, so is worthwhile IMO. :+1: